ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.ts

```diff
@@ -156,20 +123,42 @@ export class ActivityFileAnswer implements OnInit, OnDestroy {
         this.ngUnsubscribe.complete();
     }
 
-    private initUploadedFile(): void {
-        if (this.block?.answer) {
-            this.setUploadedFile({
-                fileName: this.block.answer.fileName,
-                fileSize: this.block.answer.fileSize
-            });
+    private prepareFileUploads(fileUploadRes: FileUploadResponse[], files: File[]): void {
+        const fileUploadsArray: any[] = [];
+        const fileUploadGuidsArray: string[] = [];
+        fileUploadRes.forEach((upRes, i) => {
+            fileUploadGuidsArray.push(upRes.uploadGuid);
+            fileUploadsArray.push(this.fileUploadService.uploadFile(upRes.uploadUrl, files[i]));
+        });
+
+        this.startUploadingFiles(files, fileUploadsArray, fileUploadGuidsArray);
+    }
+
+    private validateAndSubmitFileUpload(files: File[], fileUploadRes: FileUploadResponse[]): void {
+        const isValid = files.every(file => {
+            const failedLocalValidator = this.getFailedLocalValidator(file);
+            if (failedLocalValidator) {
+                this.errorMessage = failedLocalValidator.result as string;
+                this.componentBusy.emit(false);
+                return false;
+            }
+            return true;
+        });
+
+        isValid && this.prepareFileUploads(fileUploadRes, files);
+    }
+
+    private initUploadedFiles(): void {
+        if(this.block?.answer?.length > 0 && this.block?.answer[0] !== null) {
+            this.block.answer && this.uploadedFiles.push(...this.block.answer);
```

Comment:
`this.block.answer &&` is not required because it is already checked in the **if** condition.